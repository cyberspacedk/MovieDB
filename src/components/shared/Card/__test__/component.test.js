import React from 'react';
import { shallow } from 'enzyme';
import CardItem from '../component';

describe('Component: Card', () => {
  const props = {
    item: { title: 'Tittle', overview: 'Overview', poster_path: 'path' },
    handleShowMoreDetails: () => {},
    handleRemoveWatchModal: () => {},
    removeBox: true,
  };

  it('Match its snapshot. Item provided', () => {
    const wrapper = shallow(<CardItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Match its snapshot. Card without image', () => {
    const nextProps = {
      ...props,
      item: { title: 'Tittle', overview: 'Overview' },
    };
    const wrapper = shallow(<CardItem {...nextProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Match its snapshot. Card without remove box', () => {
    const nextProps = {
      ...props,
      removeBox: false,
    };
    const wrapper = shallow(<CardItem {...nextProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
