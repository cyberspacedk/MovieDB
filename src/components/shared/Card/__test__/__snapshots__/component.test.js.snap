// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Card Match its snapshot. Card without image 1`] = `
<Col
  lg={
    Object {
      "span": 6,
    }
  }
  md={
    Object {
      "span": 8,
    }
  }
  onClick={[Function]}
  sm={
    Object {
      "span": 12,
    }
  }
  xl={
    Object {
      "span": 4,
    }
  }
  xs={
    Object {
      "span": 12,
    }
  }
>
  <Card
    actions={
      Array [
        <Icon
          onClick={[Function]}
          type="delete"
        />,
      ]
    }
    className="top-margin card-film"
    cover={null}
    hoverable={true}
  >
    <Meta
      description="Overview..."
      title="Tittle"
    />
  </Card>
</Col>
`;

exports[`Component: Card Match its snapshot. Card without remove box 1`] = `
<Col
  lg={
    Object {
      "span": 6,
    }
  }
  md={
    Object {
      "span": 8,
    }
  }
  onClick={[Function]}
  sm={
    Object {
      "span": 12,
    }
  }
  xl={
    Object {
      "span": 4,
    }
  }
  xs={
    Object {
      "span": 12,
    }
  }
>
  <Card
    actions={null}
    className="top-margin card-film"
    cover={
      <img
        alt="Tittle"
        src="https://image.tmdb.org/t/p/w200path"
      />
    }
    hoverable={true}
  >
    <Meta
      description="Overview..."
      title="Tittle"
    />
  </Card>
</Col>
`;

exports[`Component: Card Match its snapshot. Item provided 1`] = `
<Col
  lg={
    Object {
      "span": 6,
    }
  }
  md={
    Object {
      "span": 8,
    }
  }
  onClick={[Function]}
  sm={
    Object {
      "span": 12,
    }
  }
  xl={
    Object {
      "span": 4,
    }
  }
  xs={
    Object {
      "span": 12,
    }
  }
>
  <Card
    actions={
      Array [
        <Icon
          onClick={[Function]}
          type="delete"
        />,
      ]
    }
    className="top-margin card-film"
    cover={
      <img
        alt="Tittle"
        src="https://image.tmdb.org/t/p/w200path"
      />
    }
    hoverable={true}
  >
    <Meta
      description="Overview..."
      title="Tittle"
    />
  </Card>
</Col>
`;
